docs/extending.txt

```diff
@@ -549,3 +549,44 @@ arguments.  Example use:
         endswith=endswith) + t
 
     site.template_hooks['page_header'].append(greeting, True, u'Nikola Tesla', endswith=u'!')
+
+Shortcodes
+==========
+
+Some (hopefully all) markup compilers support shortcodes in these forms::
+
+    {{% foo %}}  # No arguments
+    {{% foo bar %}}  # One argument, containing "bar"
+    {{% foo bar baz=bat %}}  # Two arguments, one containing "bar", one called "baz" containing "bat"
+    
+    {{% foo %}}Some text{{% /foo %}} # one argument called "data" containing "Some text"
+
+So, if you are creating a plugin that generates markup, it may be a good idea to register it as a shortcode
+in adition of a restructured text directive or markdown extension, thus making it available to all markups.
```

Comment:
in addition to a reStructuredText directive or Markdown extension, thus making it available to all input formats.